internal/impl/openai/chat_processor_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openai
+
+import (
+	"context"
+	"testing"
+
+	oai "github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai"
+	"github.com/go-faker/faker/v4"
+	"github.com/redpanda-data/benthos/v4/public/bloblang"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type mockChatClient struct {
```

Comment:
Why not use `stubClient` directly?

Reply:
I was trying to keep the tests local and self described (instead of having mocks in another file), as well as not bring in test only dependencies in the mock itself.

Is there another local pattern that ya'll tend to do for mocking? I am happy to adopt a pattern (since this does look a little funny :smile:)

Reply:
Oh, I see! You wanted to be able to overwrite various methods like `GetChatCompletions()` in tests. It makes sense. In this case, since `stubClient` is only used in tests, it should be placed in a file which ends in `_test.go`. Guess `client_test.go` or something like that.

Reply:
Thanks - done! Feels a little strange to have a `_test.go` file without tests 😄